tests/zfs-tests/tests/functional/zvol/zvol_misc/zvol_misc_common.kshlib

```diff
@@ -37,8 +37,11 @@
 #
 function udev_wait
 {
+	if ! is_linux; then
```

Comment:
It may be possible to use `block_device_wait` on all platforms now that devices can be passed to it.  I'd think it would now correctly retry in the cases where the device was open, which would be more reliably then just checking if `zvol_id` or `blkid` are running.  It's probably worth trying.

Reply:
Ok I'll give that a shot. That was my original plan until I saw what was going on in here then I started having second thoughts about touching something so purpose-built. :sweat_smile: